signer/src/transaction_coordinator.rs

```diff
@@ -390,6 +392,75 @@ where
         .map(|_| ())
     }
 
+    /// Constructs a BitcoinPreSignRequest from the given transaction package and
+    /// sends it to the signers. Waits for acknowledgments from the signers until
+    /// the threshold is met or a timeout occurs.
+    /// If the signal stream closes unexpectedly, triggers a shutdown.
+    #[tracing::instrument(skip_all)]
+    async fn construct_and_send_bitcoin_presign_request(
+        &mut self,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
+        pending_requests: &SbtcRequests,
+        transaction_package: &[utxo::UnsignedTransaction<'_>],
+    ) -> Result<(), Error> {
+        // Create the BitcoinPreSignRequest from the transaction package
+        let sbtc_requests = BitcoinPreSignRequest {
+            request_package: transaction_package
+                .iter()
+                .map(|tx| (&tx.requests).into())
+                .collect(),
+            fee_rate: pending_requests.signer_state.fee_rate,
+            last_fees: pending_requests.signer_state.last_fees.map(Into::into),
+        };
+        let presign_ack_filter = {
+            let target_tip = *bitcoin_chain_tip;
+            move |event: &SignerSignal| {
+                matches!(
+                    event,
+                    SignerSignal::Event(SignerEvent::TxSigner(TxSignerEvent::MessageGenerated(
```

Comment:
I think you still need to also include `P2PEvent::MessageReceived(Signed { etc. })`

Reply:
hm, yeah that's a bit worrying... even counting from 1 I guess it should fail if the tests are testing with several signers and a threshold greater than 1.

Reply:
With 2 out of 3, considering it starts from 1 and the coordinator signer it should reach the required 2 acks I guess

Reply:
Yeah I guess so. I thought I've seen tests running like 5-7 though. But maybe not for the coordinator.

Reply:
fixed in https://github.com/stacks-network/sbtc/pull/956/commits/d86363dbb1f0164a5944ab6ac71c399390691dcb